javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/ReferenceMatcher.java

```diff
@@ -192,7 +192,7 @@ private static void collectMethodsFromTypeHierarchy(
     List<Mismatch> mismatches = Collections.emptyList();
 
     for (Reference.Flag flag : reference.getFlags()) {
-      if (!flag.matches(typeOnClasspath.getModifiers())) {
+      if (!flag.matches(typeOnClasspath.getActualModifiers(false))) {
```

Comment:
I believe it is better to use `getModifiers` for fields and methods because the only extra bit that `getActualModifiers` returns for fields is `ACC_DEPRECATED` which shouldn't be needed for us. For classes I changed it to use `getActualModifiers` because a protected inner class is really a public class, we shouldn't really care about the modifiers in inner class attribute as vm doesn't use it for access checks anyway.